flexget/plugins/output/plugin_decompress.py

```diff
@@ -86,36 +86,30 @@ def open_archive(self, entry):
                 error_message = 'Bad ZIP file: %s' % archive_path
                 log.error(error_message)
                 entry.fail(error_message)
-                return None
-            except Exception as e:
-                log.error('Failed to open ZIP: %s (%s)' (archive_path, e))
-                entry.fail(e)
-                return None
-        elif rarfile:
-            if rarfile.is_rarfile(archive_path):
-                try:
-                    archive = rarfile.RarFile(rarfile=archive_path)
-                    log.debug('Successfully opened RAR: %s' % archive_path)
-                except rarfile.RarWarning as e:
-                    log.warn('Nonfatal error: %s (%s)' % (archive_path, e))
-                except rarfile.NeedFirstVolume:
-                    log.error('Not the first volume: %s' % archive_path)
-                    return None
-                except rarfile.NotRarFile:
-                    log.error('Not a RAR file: %s' % archive_path)
-                    return None
-                except Exception as e:
-                    log.error('Failed to open RAR: %s (%s)' (archive_path, e))
-                    entry.fail(e)
-                    return None
+            except Exception as error:
+                error_message = 'Failed to open ZIP: %s (%s)' (archive_path, error)
+                log.error(error_message)
+                entry.fail(error_message)
+        elif rarfile and rarfile.is_rarfile(archive_path):
+            try:
+                archive = rarfile.RarFile(rarfile=archive_path)
+                log.debug('Successfully opened RAR: %s' % archive_path)
+            except rarfile.BadRarfile:
+                error_message = 'Bad RAR file: %s' % archive_path
+                log.error(error_message)
+                entry.fail(error_message)
+            except rarfile.RarWarning as error:
+                log.warn('Nonfatal error: %s (%s)' % (archive_path, error))
```

Comment:
I still don't understand how this is a nonfatal error. If it happens, `archive` will be None. Certainly that's fatal?